### ISSUE TITLE
Incompatible default openssl versions (1.1.0f) for python 3.5

### ISSUE THREAD

**Issue by KPrasch** [nucypher/nucypher#560]
Needs investigation - Results in the following exception, which appears to occur before certificate verification.

`[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:720)`

**Comment by michwill** [comment]
This is happening in Debian Stable (9) and standard python3.5 over there (which uses OpenSSL-1.1.0f). Can be checked using `debian:stretch` docker container, for example

**Comment by KPrasch** [comment]
TLS 1.3 ciphers are enabled by default in Open SSL 1.1.1 - which comes with python 3.6 out of the box.

See related PR: <https://github.com/twisted/twisted/pull/774>
Ticket: <https://twistedmatrix.com/trac/ticket/9128>

This might be a won't fix. At least for now the fix is to use python 3.6+ or a version of python < 3.6 compiled with an  OpenSSL > 1.1.0.

**Comment by tuxxy** [comment]
This probably isn't the right issue, but I guess we're blocked on upstream waiting for a Twisted release that includes TLSv1.3?
